sci-libs/openblas/openblas-0.3.6.ebuild

```diff
@@ -11,34 +11,56 @@ SRC_URI="https://github.com/xianyi/OpenBLAS/tarball/v${PV} -> ${P}.tar.gz"
 LICENSE="BSD"
 SLOT="0"
 KEYWORDS="~amd64 ~x86 ~amd64-linux ~x86-linux ~x64-macos ~x86-macos"
-IUSE="dynamic openmp pthread serial static-libs eselect-ldso index-64bit"
-REQUIRED_USE="?? ( openmp pthread serial )"
+IUSE="dynamic eselect-ldso index-64bit np2 np4 np8 np16 np32 openmp threads"
+REQUIRED_USE="?? ( np2 np4 np8 np16 np32 )"
 
```

Comment:
I am still not clear how 'serial' & 'static-libs' use flags trigger in the ebuild. Maybe i am missing something. Could you check again? I keeped PR that removes them. If really i am missing something please inform me. I would re-add them.

Also i believe REQUIRED_USE="?? ( openmp pthread serial )" is not necessary.
If user has 'openmp' global flag in make.conf and try to compile openblas with 'pthread' .This is hard-blocked with REQUIRED_USE.

We already have below code block that more flexible than REQUIRED_USE,

I re-ordered if clauses because of that. It relies first pthread (local flag) then openmp (global flag).

```
        if use pthread; then
                flags+=( USE_THREAD=1 USE_OPENMP=0 )
        elif use openmp; then
                flags+=( USE_THREAD=1 USE_OPENMP=1 )
                # Check OpenMP support on compiler.
                tc-check-openmp
        else
                flags+=( USE_THREAD=0 )
        fi
```

Reply:
> I am still not clear how 'serial' & 'static-libs' use flags trigger in the ebuild. Maybe i am missing something. Could you check again? I keeped PR that removes them. If really i am missing something please inform me. I would re-add them.

Oh yes, serial and static-libs are not needed.

> Also i believe REQUIRED_USE="?? ( openmp pthread serial )" is not necessary.
> If user has 'openmp' global flag in make.conf and try to compile openblas with 'pthread' .This is hard-blocked with REQUIRED_USE.

This is not a problem, as you can always set

`sci-libs/openblas -openmp` in the `package.use` to override it.

> We already have below code block that more flexible than REQUIRED_USE,
> 
> I re-ordered if clauses because of that. It relies first pthread (local flag) then openmp (global flag).
> 
> ```
>         if use pthread; then
>                 flags+=( USE_THREAD=1 USE_OPENMP=0 )
>         elif use openmp; then
>                 flags+=( USE_THREAD=1 USE_OPENMP=1 )
>                 # Check OpenMP support on compiler.
>                 tc-check-openmp
>         else
>                 flags+=( USE_THREAD=0 )
>         fi
> ```

No, it is cannot replace REQUIRED_USE.   The build system cannot build both openmp and pthread, and REQUIRED_USE is the only to express it.